src/main/java/org/folio/rest/impl/CheckinReceivePiecesHelper.java

```diff
@@ -737,4 +726,19 @@ private boolean isMissingLocation(PoLine poLine, Piece piece) {
       }
     });
   }
+
+  CompletionStage<List<CompletableFuture<Void>>> removeForbiddenEntities(List<PoLine> poLines, Map<String, Map<String, T>> pieces) {
+    Map<String, List<PoLine>> poLinesGroupedByOrderId = poLines.stream().collect(groupingBy(PoLine::getPurchaseOrderId));
+    String query = buildQuery(convertIdsToCqlQuery(new ArrayList<>(poLinesGroupedByOrderId.keySet())), logger);
+    String url = String.format(GET_PURCHASE_ORDERS, Integer.MAX_VALUE, 0, query, lang);
+    return handleGetRequest(url, httpClient, ctx, okapiHeaders, logger)
+      .thenApply(orders -> orders.mapTo(PurchaseOrders.class).getPurchaseOrders().stream().map(order -> protectionHelper.isOperationRestricted(order.getAcqUnitIds(), ProtectedOperationType.UPDATE)
+        .exceptionally(t -> {
+          poLinesGroupedByOrderId.get(order.getId())
+            .forEach(line -> Objects.requireNonNull(pieces.computeIfPresent(line.getId(), (s, stringCheckInPieceMap) -> pieces.remove(line.getId()))).keySet()
```

Comment:
What is the benefit of `pieces.computeIfPresent` usage instead of just `pieces.remove`?
`Objects.requireNonNull` throws NPE exception which actually terminates entire flow. What is the reason to use it?